awpy/analytics/map_control.py

```diff
@@ -0,0 +1,403 @@
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
+
+    Args:
+        map_name (str): Map used for find_closest_area and
+            relevant tile neighbor dictionary
+        current_player_data (FrameTeamMetadata): Object containing team metadata
+            (player positions, etc.). Expects extract_team_metadata output format
+
+    Returns: FrameMapControlValues object containing each team's map control values
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    neighbors_dict = graph_to_tile_neighbors(list(NAV_GRAPHS[map_name].edges))
+
+    t_tiles = [
+        find_closest_area(map_name, i)["areaId"]
+        for i in current_player_data.t_metadata.alive_player_locations
+    ]
+    ct_tiles = [
+        find_closest_area(map_name, i)["areaId"]
+        for i in current_player_data.ct_metadata.alive_player_locations
+    ]
+
+    return _frame_tile_map_control_values(map_name, ct_tiles, t_tiles, neighbors_dict)
+
+
+def calc_frame_map_control_values(
+    map_name: str,
+    frame: GameFrame,
+) -> FrameMapControlValues:
+    """Calculate tile map control values for each team given frame object.
+
+    Values are allocated to tiles depending on how many tiles are between it
+    and the source tile (aka tile distance). The smaller the tile distance,
+    the close the tile's value is to 1. The tile distance can get as high as
+    the max_depth argument, which is 10 as default. If the max_depth is kept
+    as default 10, this means the BFS search will stop once a tile
+    distance of 10 is reached.
+
+    Args:
+        map_name (str): Map used for find_closest_area and
+            relevant tile neighbor dictionary
+        frame (GameFrame): Awpy frame object for map control calculations
+
+    Returns: FrameMapControlValues object containing each team's map control values
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    return calc_parsed_frame_map_control_values(
+        map_name=map_name,
+        current_player_data=extract_teams_metadata(frame),
+    )
+
+
+def _extract_team_metadata_helper(
+    side_data: TeamFrameInfo,
+) -> TeamMetadata:
+    """Helper function to parse player locations in given side_data.
+
+    Args:
+        side_data (TeamFrameInfo): Object with metadata for side's players.
+
+    Returns: TeamMetadata with metadata on team's players
+    """
+    coords = ("x", "y", "z")
+    alive_players: list[list[float]] | list = [
+        [player[dim] for dim in coords]
+        for player in side_data["players"] or []
+        if player["isAlive"]
+    ]
+
+    return TeamMetadata(alive_player_locations=alive_players)
```

Comment:
You should probably make use of the `PlayerPositions` alias here, right?

`alive_players: list[PlayerPosition] = ...`

Reply:
Yeah I missed this. Thanks for this catch - addressed in next commit.